src/puppetlabs/puppetdb/import.clj

```diff
@@ -36,17 +60,32 @@
           ;;   the list of nodes that we submitted and the output of that query
           (command-fn :replace-catalog
                       (:replace_catalog command-versions)
-                      (utils/read-json-content tar-reader)))
+                      (extract-certname "catalogs" path)
+                      (-> tar-reader
+                          utils/read-json-content
```

Comment:
This is a hack around the awkward TAR lib we're using. It gives 1 input stream that is "marked" for entries in the tar. What we really want here is a "substream" kind of thing where we can pass Stockpile a stream that is a slice into that big stream, i.e. byte 1,000 through 75,000. AFAIK, that's not a thing and if it was, we'd still need to make some significant changes here to make it work. I just hacked around it with a pretty inefficient call to generate-string and converted it a ByteArrayInputStream